### PR TITLE
Support automatic cookie removal on logout

### PR DESCRIPTION
**Pull request by CSchulz**
Support of cookie destroying on logout by adding "remove_cookies: { 'cookieName1', 'cookieName2':  { 'path': '/', 'domain': null } }" in your configuration e.g.

```
$app['security.firewalls'] = array(
[...]
        'logout' => array(
            'logout_path' => '/user/logout',
            'remove_cookies' => array('cookieName1', 'cookieName2' => array('path' => '/', 'domain' => null))
        ),
[...]
);
```

Based on discussion https://groups.google.com/d/topic/silex-php/828UcSi7bks/discussion

### CODE REVIEW

**Comment by DerManoMann** [comment]
:+1:

**Comment by fabpot** [comment]
Can you update the documentation for this new feature? Thanks.

File: src/Silex/Provider/SecurityServiceProvider.php [hunk 1]
```diff
@@ -469,6 +470,32 @@ public function register(Application $app)
 
                 $listener->addHandler(new SessionLogoutHandler());
 
+                $removeCookies = isset($options['remove_cookies']) ? $options['remove_cookies'] : null;
+                if (null !== $removeCookies) {
```
Comment (GromNaN) [inline src/Silex/Provider/SecurityServiceProvider.php]: The `$removeCookies` is not necessary at all.
Should be directly `if (isset($options['remove_cookies'])) {` and then `foreach ($options['remove_cookies'])` ...

File: src/Silex/Provider/SecurityServiceProvider.php [hunk 2]
```diff
@@ -469,6 +470,32 @@ public function register(Application $app)
...
+                    $cookies = array();
+
+                    foreach ($removeCookies as $key => $value) {
+                        if (is_array($value)) {
+                            // Check for existence of method parameters otherwise
+                            // use default values given by ResponseHeaderBag:clearCookie()
+                            if (!isset($value['path'])) {
+                                $value['path'] = '/';
```
Comment (GromNaN) [inline src/Silex/Provider/SecurityServiceProvider.php]: It would be nice if the `CookieClearingLogoutHandler` could make this check itself.

**Comment by CSchulz** [comment]
I will update the PR soon.

**Comment by fabpot** [comment]
@CSchulz Any news?

File: src/Silex/Provider/SecurityServiceProvider.php [hunk 3]
```diff
@@ -469,6 +470,32 @@ public function register(Application $app)
...
+                                $value['domain'] = null;
+                            }
+
+                            $cookies[$key] = $value;
+                        } else {
+                            // Possibility to skip cookie values and use default
+                            // values given by ResponseHeaderBag:clearCookie()
+                            $cookies[$value] = array('path' => '/', 'domain' => null);
+                        }
+                    }
```
Comment (fabpot) [inline src/Silex/Provider/SecurityServiceProvider.php]: I would remove all this logic and document how to pass a valid `$cookies` array to `CookieClearingLogoutHandler `.

**Comment by ragboyjr** [comment]
Any update on this?

**Comment by fabpot** [comment]
Closing this old PR as there is no more activity.

**Pull request closed without merging**